src/parser/Parser.ts

```diff
@@ -2868,6 +2959,18 @@ export class References {
     }
     private _interfaceStatementLookup: Map<string, InterfaceStatement>;
 
+    /**
+     * A collection of full expressions. This excludes intermediary expressions.
+     *
+     * Example 1:
+     * `a.b.c` is composed of `a` (variableExpression)  `.b` (DottedGetExpression) `.c` (DottedGetExpression)
+     * This will only contain the final `.c` DottedGetExpression because `.b` and `a` can both be derived by walking back from the `.c` DottedGetExpression.
+     *
+     * Example 2:
+     * `name.space.doSomething(a.b.c)` will result in 2 entries in this list. the `CallExpression` for `doSomething`, and the `.c` DottedGetExpression.
+     */
+    public expressions = new Set<Expression>();
```

Comment:
This is the main point of the change. It contains a list of all full expressions found in the parsed file. Can be very useful for enum validations, other property validations, etc.

Reply:
Why a set and not an array?

Reply:
Why not an array?

Reply:
I'm using the `Set` to prevent duplicates, and also I remove elements occasionally.